src/utils/wrappers.py

```diff
@@ -0,0 +1,343 @@
+    """Road Imagery Class: Wrapper for ee.Image."""
+
+    def __init__(self, link, bands):
+        """Initialize the Imagery Class.
+
+        Parameters
+        ----------
+        link : str
+            ee.Image link
+        bands : list[str], optional
+            Name of bands to select. The default is None to take all bands.
+
+        Returns
+        -------
+        None.
+
+        """
+        self.bands = bands
+        self.imagery = ee.Image(link).select(self.bands)
+
+    def get_bands(self, geometry):
+        """.
+
+        Parameters
+        ----------
+        geometry : ee.Geometry
+            Geometry to clip to
+
+
+        Returns
+        -------
+        ee.Image
+            Image clipped to the geometry
+
+        """
+        return self.imagery.clip(geometry).unmask(0, False)
+
+
+class TropomiImagery(CollectionClass):
+    """."""
+
+    def __init__(self, link, bands=None, before_range=(0, 'day'),
+                 after_range=(7, 'day')):
+        """.
+
+        Parameters
+        ----------
+        link : TYPE
+            DESCRIPTION.
+        bands : TYPE, optional
+            DESCRIPTION. The default is None.
+        before_range : TYPE, optional
+            DESCRIPTION. The default is (0,'day').
+        after_range : TYPE, optional
+            DESCRIPTION. The default is (7,'day').
+
+        Returns
+        -------
+        None.
+
+        """
+        super(TropomiImagery, self).__init__(link, bands)
+        self.before = before_range
+        self.after = after_range
+
+    def get_bands(self, date, geometry):
+        """."""
+        return self.get_clipped_range_bands(geometry,
+                                            date.advance(*self.before),
+                                            date.advance(*self.after))
+
+
+class MultiSpectralImagery(CollectionClass):
+    """."""
+
+    def __init__(self, link, start_date, end_date, bands, scale):
+        """.
+
+        Parameters
+        ----------
+        link : TYPE
+            DESCRIPTION.
+        start_date : TYPE
+            DESCRIPTION.
+        end_date : TYPE
+            DESCRIPTION.
+        bands : TYPE
+            DESCRIPTION.
+        scale : TYPE
+            DESCRIPTION.
+
+        Returns
+        -------
+        None.
+
+        """
+        super(MultiSpectralImagery, self).__init__(link, bands)
+
+        self.imagery = self.imagery.filterDate(start_date, end_date)
+        self.imagery = self.imagery.filter(ee.Filter.notNull(
+            ['collectionStartTime']))
+        self.size = self.imagery.size().getInfo()
+        self.imagery_list = self.imagery.toList(self.size)
+        print("Base Imagery has %i elements" % self.size)
+        self.scale = scale
+
+    def __getitem__(self, i):
+        """.
+
+        Parameters
+        ----------
+        i : int
+            index of element to return
+
+        Returns
+        -------
+        ee.Image
+
+        """
+        assert i < len(self)
```

Comment:
Good check! Can you add an error message to this?

assert i < len(self), "MultiSpectral failed because blablabla"